### PR TITLE
Confirm Publish

### PR DESCRIPTION
**Pull request by bkendall**
In order to make the `.publishTo*` methods ensure they _actually_ publish when they resolve, I have to use a "confirm queue" through `amqplib` to make sure that things are confirmed when published. Requires setting up a second channel (which was a thought originally anyway), but does make for better reliability when publishing messages.

### CODE REVIEW

**Comment by anandkumarpatel** [comment]
code LGTM, only concern i would have is latency. If your confident merge it!

**Comment by bkendall** [comment]
Changed the strategy. Instead of confirming every publish, we can confirm that everything that had been published is confirmed when we close the connection. This should prevent any performance issues when publishing

File: src/rabbitmq.js [hunk 1]
```diff
@@ -108,6 +103,18 @@ class RabbitMQ {
         this.log.info('created channel')
         this.channel = channel
         this.channel.on('error', this._channelErrorHandler.bind(this))
+
+        this.log.info('creating publishing channel')
+        return Promise.resolve(this.connection.createConfirmChannel())
```
Comment (anandkumarpatel) [inline src/rabbitmq.js]: does this still need to be created here? or can you create it at close as well?

Comment (bkendall) [inline src/rabbitmq.js]: no, need to create both channels still when it's connected

File: src/rabbitmq.js [hunk 2]
```diff
@@ -132,7 +139,8 @@ class RabbitMQ {
       }
       const stringContent = JSON.stringify(content)
       const bufferContent = new Buffer(stringContent)
-      return Promise.resolve(this.channel.sendToQueue(queue, bufferContent))
+      return Promise
+        .resolve(this.publishChannel.sendToQueue(queue, bufferContent))
```
Comment (anandkumarpatel) [inline src/rabbitmq.js]: quick question. I was looking through the docs and from the example they gave it looks like it waits for the message to be ack'd before it completes?
```
require('amqplib/callback_api').connect(function(err, c) {
  c.createConfirmChannel(function(err, ch) {
    ch.sendToQueue('foo', new Buffer('foobar'), {}, function(err, ok) {
      if (err !== null) console.warn('Message nacked!');
      else console.log('Message acked');
    });
  });
});
```

do you know if that means it wait for the jobs `ack` or is that some other kind of `ack` it waits for?

http://www.squaremobius.net/amqp.node/channel_api.html#confirmchannel

Comment (bkendall) [inline src/rabbitmq.js]: it's just that the server acknowledges it, not that the message is passed through and handled by some worker...

**Comment by anandkumarpatel** [comment]
LGTM

**Pull request merged**